examples/normalizer/Normalize.py

```diff
@@ -0,0 +1,177 @@
+            jobs = self.jobs
+
+        next_device = jobs.pop()
+        log.msg('Selecting next device: %s' % next_device)
+
+        # If I ping, return me.
+        if next_device.is_reachable():
+            log.msg('PING [SUCCESS]: %s' % next_device)
+            return next_device
+        # Otherwise return None and store me as an error.
+        else:
+            msg = 'PING [FAILURE]: %s' % next_device
+            log.msg(msg)
+            print "Failed to ping host {}".format(next_device)
+            if self.verbose:
+                print msg
+            return None
+
+    def errback(self, failure, device):
+        print "Error in getRouterDetails for device {}\n{}".format(
+            device,
+            failure.getTraceback()
+        )
+
+
+def validateRouterDetails(result):
+    print "Validating router details"
+    devicesToCorrect = []
+
+    for device, results in result.items():
+        routers[device].results = results
+        routers[device].validate()
+        routers[device].normalize()
+        if routers[device].normalizeRequired:
+            devicesToCorrect.append(device)
+
+    return devicesToCorrect or None
+
+
+class normalizeRouters(ReactorlessCommando):
+    def to_cisco(self, dev, commands=None, extra=None):
+        dev_commands = routers[dev.nodeName].commands
+        # self.commands = dev_commands
+        # print "Device {}: Executing Commands:\n{}".format(
+        #       dev.nodeName, dev_commands)
+        return dev_commands
+
+    def from_cisco(self, results, device, commands=None):
+        commands = commands or self.commands
+
+        log.msg('Received %r from %s' % (results, device))
+        self.store_results(device, self.map_results(commands, results))
+
+    def errback(self, failure, device):
+        print "Error in normalizeRouters for device {}\n{}".format(
+            device,
+            failure.getTraceback()
+        )
+
+
+def initiateRouterNormalization(devices):
+    # log.startLogging(sys.stdout, setStdout=False)
+    if devices is not None:
+        print "Normalizing {} devices ({})".format(
+            len(devices),
+            " ".join(devices)
+        )
+        devicesToNormalize = []
+        for device in devices:
+            if routers[device].normalizeRequired:
+                devicesToNormalize.append(device)
+        return normalizeRouters(devicesToNormalize).run()
+    else:
+        print "No devices need to be normalized".format()
+    return None
+
+
+def stop_reactor(result):
+    if reactor.running:
+        reactor.stop()
+        return result
+
+def main():
+    nd = NetDevices()
+
+    global device_list
+    device_list = []
+    global routers
+    routers = {}
+
+    try:
+        stateFile = open("routerstate.json", "r")
+        data = stateFile.read()
+        stateFile.close()
+        routers = jsonpickle.decode(data)
+        data = None
+    except Exception as e:
+        print("Failed to open routerstate.json")
```

Comment:
This is always a tricky one.

In python 2, unless you state `from __future__ import print_function` then really all this is doing is printing a tuple though since it has only one item it'll print index 0. Can be confusing if adding an extra string to it.

It's also a bit inconsistent since most of the rest uses regular statement print. 

L121 doesn't need `as e` because there's really nothing being done with it here.